hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -418,7 +418,8 @@
     {
       name: chip_jtag_csr_rw
       desc: '''
-            Verify accessibility of CSRs as indicated in the RAL specification.
+            Verify accessibility of CSRs as indicated in the RAL specification, including CSRs in
+            other blocks.
```

Comment:
@sriyerg this test is not very clear to me. After review all the test entries here, I think we want to test all CSRs in the chip through jtag, right?

Reply:
Seems like there's a bit of confusion ongoing here that mixes the chip_jtag_csr_rw and chip_rv_dm_cpu_debug_mem tests.

For the chip_jtag_csr_rw test, we discussed that we want to test the access to the Debug Module Registers (https://github.com/pulp-platform/riscv-dbg/blob/master/doc/debug-system.md#debug-module-registers) through JTAG by doing a read-write round-trip test. Since most of these registers have side effects, we tried to pick one or two registers which are known to be "plain" registers.

How about a text like this:

```
- Write RW type Debug Module Registers (such as progbuf0 and progbuf1) through JTAG with a random value.
- Read back to check their values for correctness.
```

Please also rename the test from chip_jtag_csr_rw to e.g. chip_jtag_debug_module_register_rw to make it clear what type of register we're talking about.

Reply:
@imphil I have confirmed with @sriyerg offline.
The intent of this test is to reuse `chip_csr_rw` to use jtag to test all the CSRs in the chip. so this will also test SBA functionality.

Reply:
Thanks for checking. In this case, please reword to clearly indicate what is meant by "CSR" and which "RAL specification" is meant (is there one for rv_dm?).

Reply:
Updated the wording. It should be all the CSRs in the chip. If it's only rv_dm CSR, then we should test that in block-level. top-level is to cover interaction with other blocks.

Reply:
The rv_dm section of the testplan originally was written with likely incorrect assumptions. The point of the testplan review was to fix those assumptions and update it to reflect the current state of the design as it is integrated into the chip. I assume that's done (IIRC I was unable to make it). I will go with the fact that @weicaiyang has updated the testplan based on your (@imphil & @GregAC) comments and it is now up-to-date.